packages/core/src/utils/parsers.tsx

```diff
@@ -70,6 +73,33 @@ export const isDidCommInvitation = (url: string) => {
   return false
 }
 
+export const parseMediatorInvitation = (url: string): Record<string, any> | null => {
+  if (!url.includes('c_i=')) return null
+
+  try {
+    const { c_i: encoded } = parse(url.split('?')[1] || '')
+    if (typeof encoded !== 'string') return null
+
+    let base64 = encoded.replace(/-/g, '+').replace(/_/g, '/')
+    while (base64.length % 4 !== 0) base64 += '='
+
+    const decoded = Buffer.from(base64, 'base64').toString('utf-8')
+    return JSON.parse(decoded)
+  } catch (e) {
+    return null
+  }
+}
+
+export const isMediatorInvitation = (url: string): boolean => {
+  const invitation = parseMediatorInvitation(url)
+
+  if (invitation) {
+    return typeof invitation.label === 'string' && invitation.label.toLowerCase().includes('mediator')
```

Comment:
So the only way to tell if it's a mediator is if the label contains the word mediator? That might be a bit brittle